packages/client/src/comp/states/Base.tsx

```diff
@@ -23,75 +26,148 @@ type Props<Info> = React.PropsWithChildren<{
+  const [ ignoreRegistrationStatus, setIgnoreRegistrationStatus ] = React.useState<boolean>(false)
+  const [ isOpen, setIsOpen ] = React.useState<boolean>(false)
+  const [ modalContext, setModalContext ] = React.useState<'formSubmit'|'click'>('click')
 
-  const nameRef = useControlRef<Input>()
-  const birthdateRef = useControlRef<Input>()
-  const emailRef = useControlRef<Input>()
-  const phoneRef = useControlRef<Input>()
-  const mailingRef = useControlRef<Input>()
   if (!locale || !isImplementedLocale(locale) || !contact) return null
 
   const uspsAddress = address ? address.fullAddr : null
   const { city, county, otherCities, latLong } = locale
 
-  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
-    event.persist()  // allow async function call
-    event.preventDefault()
-
+  const stateInfo = (): StateInfo | null => {
     if (!address || !uspsAddress || !contact) {
       toast.error('Please fill all the required fields')
-      return
+      return null
     }
 
-    const baseInfo: StatelessInfo = {
+    const base: StatelessInfo = {
       city: contact.city ?? city,
       county: contact.county ?? county,
       otherCities,
       latLong,
       oid,
-      name: nameRef.value() || '',
-      birthdate: birthdateRef.value() || '',
-      email: emailRef.value() || '',
-      mailingAddress: mailingRef.value() || '',
-      phone: phoneRef.value() || '',
+      name: `${fields.firstName.value} ${fields.lastName.value}`,
+      birthdate: fields.birthdate.value,
+      email: fields.email.value,
+      mailingAddress: fields.mailing.value,
+      phone: fields.telephone.value,
       uspsAddress,
       contact,
       address,
     }
 
-    const info = enrichValues(baseInfo)
+    const info = enrichValues(base)
+
     if (!info) {
       // Do not dismiss previous errors which may give more details on bad fields
       toast.error('Please fill all the required fields in the right formats')
-      return
     }
+
+    return info
+  }
+
+  async function checkRegistration(
+    id: 'firstName' | 'lastName' | 'birthdate',
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) {
+    e.preventDefault()
+    e.persist() // allow async function call
+    updateField(id, e.currentTarget.value)
+    if (canCheckRegistration() && !ignoreRegistrationStatus) {
+      const {
+        firstName, lastName, birthdate,
+      } = fields
+      setRegistrationStatus('Loading')
+
+      const result = await client.isRegistered({
+        firstName: firstName.value,
+        lastName: lastName.value,
+        birthdate: birthdate.value,
+        city: address?.city ?? '',
+        postcode: address?.postcode ?? '',
+        stateAbbr: address?.stateAbbr ?? '',
+        street: address?.street ?? '',
+        streetNumber: address?.streetNumber ?? '',
+      })
+      if (result.type === 'data') {
+        setRegistrationStatus(result.data ?? 'Error')
```

Comment:
Nice observation, I think null coalescing `result.data` is no longer necessary since `isRegistered` doesn't return `null` anymore. And if anything wrong happens `result.type` will be `error`, which is already handled as 'Error'